kpm/lib/kpm/base_artifact.rb

```diff
@@ -44,12 +44,22 @@ def nexus_remote(overrides = {}, ssl_verify = true, logger = nil)
         # overrides typically comes from the kpm.yml where we expect keys as String
         overrides_sym = (overrides || {}).each_with_object({}) { |(k, v), h| h[k.to_sym] = v }
         nexus_config = nexus_defaults.merge(overrides_sym)
-        KPM::NexusFacade::RemoteFactory.create(nexus_config, ssl_verify, logger)
+
+        # Choose the appropriate facade based on the repository URL
+        url = nexus_config[:url]
+        if url&.start_with?('https://oss.sonatype.org')
```

Comment:
Maybe the name is confusing but `NexusFacade` should be the one dispatching to the right backend client implementation (GitHub, Cloudsmith, Sonatype or now Maven Central).